Apple-TV/Playback/VLCFullscreenMovieTVViewController.m

```diff
@@ -831,28 +832,25 @@ - (void)mediaPlayerStateChanged:(VLCMediaPlayerState)currentState
     }
 }
 
-- (void)displayMetadataForPlaybackController:(VLCPlaybackController *)controller
-                                       title:(NSString *)title
-                                     artwork:(UIImage *)artwork
-                                      artist:(NSString *)artist
-                                       album:(NSString *)album
-                                   audioOnly:(BOOL)audioOnly
+- (void)displayMetadataForPlaybackController:(VLCPlaybackController *)controller metadata:(VLCMetaData *)metadata
 {
+    NSLog(@"%@",metadata);
+    NSString *title = metadata.title;
+    NSString *artist = metadata.artist;
     self.titleLabel.text = title;
-
-    if (audioOnly) {
+    if (metadata.isAudioOnly) {
         self.audioArtworkImageView.image = nil;
         self.audioDescriptionTextView.hidden = YES;
         [self stopAudioDescriptionAnimation];
 
-        if (artist != nil && album != nil) {
+        if (artist != nil && metadata.albumName != nil) {
```

Comment:
consider caching the album name as you will call this property 3 times, so you could save 2 selector dispatches.